internal/manifest/oci_index.go

```diff
@@ -93,19 +100,49 @@ func (index *OCI1IndexPublic) ChooseInstance(ctx *types.SystemContext) (digest.D
 				Variant:      d.Platform.Variant,
 			}
 			if platform.MatchesPlatform(imagePlatform, wantedPlatform) {
-				return d.Digest, nil
+				if matchedDigest == "" {
+					matchedDigest = d.Digest
+				}
+				if compressionAnnotation == "" {
+					return d.Digest, nil
+				} else {
+					if _, ok := d.Annotations[compressionAnnotation]; ok {
```

Comment:
Hum, should this enforce that the value is `"true"`, and fail on other values?

I think it should, we have _one_ chance to enforce all creators to use that syntax correctly. We can always relax this later.

OTOH “be liberal in what you accept” is a long-standing idea with some attraction to it, so I don’t want to _completely_ dismiss it. What do others think?

Reply:
I agree lets be strict in the beginning until we are forced to loosen.